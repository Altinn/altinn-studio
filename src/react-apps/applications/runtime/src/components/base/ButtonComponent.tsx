import * as React from 'react';
import { connect } from 'react-redux';
import { makeGetValidationsSelector } from '../../selectors/getValidations';
import { IValidations } from '../../types/global';
import { canFormBeSaved, getErrorCount } from '../../utils/validation';
import FormDataActions from './../../features/form/data/actions/index';
import { IAltinnWindow, IRuntimeState } from './../../types';
export interface IButtonProvidedProps {
  id: string;
  text: string;
  disabled: boolean;
  handleDataChange: (value: any) => void;
  unsavedChanges: boolean;
  formDataCount: number;
}

export interface IButtonProps extends IButtonProvidedProps {
  validations: IValidations;
}

export interface IButtonState { }

export class ButtonComponentClass extends React.Component<IButtonProps, IButtonState> {
  public renderSubmitButton = () => {
    const disabled = (getErrorCount(this.props.validations) > 0) || this.props.unsavedChanges;
    return (
      <button
        type='submit'
        className={disabled ? 'a-btn a-btn-success disabled' : 'a-btn a-btn-success'}
        onClick={this.submitForm}
        disabled={disabled}
        id={this.props.id}
        style={{ marginBottom: '0' }}
      >
        {this.props.text}
      </button>
    );
  }

  // TODO: Remove saveButton and functions (and sagas) when we have implemented automatic save.
  public renderSaveButton = () => {
    const disabled = !this.props.unsavedChanges || !canFormBeSaved(this.props.validations);
    return (
      <button
        type='submit'
        className={disabled ?
          'a-btn a-btn-success disabled' : 'a-btn a-btn-success'}
        onClick={this.saveFormData}
        disabled={disabled}
        id='saveBtn'
        style={{ marginBottom: '0' }}
      >
        Lagre
      </button>
    );
  }

  public saveFormData = () => {
    const { org, app, instanceId } = window as Window as IAltinnWindow;
    FormDataActions.submitFormData(
      `${window.location.origin}/${org}/${app}/api/${instanceId}`,
    );
  }

  public submitForm = () => {
    const {org, app, instanceId } = window as Window as IAltinnWindow;
    FormDataActions.submitFormData(
      `${window.location.origin}/${org}/${app}/api/${instanceId}`,
      'Complete',
    );
  }

  public render() {
    return (
      <div className='row'>
        <div className='a-btn-group' style={{ marginTop: '3.6rem', marginBottom: '0' }}>
          {this.renderSaveButton()}
          {this.renderSubmitButton()}
        </div>
      </div>
    );
  }
}

const makeMapStateToProps = () => {
  const GetValidations = makeGetValidationsSelector();
  const mapStateToProps = (state: IRuntimeState, props: IButtonProvidedProps): IButtonProps => {
    return {
      validations: GetValidations(state),
      ...props,
    };
  };
  return mapStateToProps;
};

export const ButtonComponent = connect(makeMapStateToProps)(ButtonComponentClass);
