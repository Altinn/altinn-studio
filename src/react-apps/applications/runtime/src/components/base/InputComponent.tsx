import classNames = require('classnames');
import * as React from 'react';
import '../../styles/shared.css';

export interface IInputProps {
  id: string;
  readOnly: boolean;
  required: boolean;
  formData: any;
  handleDataChange: (value: any) => void;
  isValid?: boolean;
  type?: string;
}

export interface IInputState {
  value: string;
}

export class InputComponent
  extends React.Component<IInputProps, IInputState> {

  constructor(_props: IInputProps, _state: IInputState) {
    super(_props, _state);
    this.state = {
      value: _props.formData ? _props.formData : '',
    };
  }

  public onDataChanged = (e: any) => {
    this.setState({
      value: e.target.value,
    });
  }

  public onDataChangeSubmit = () => {
    this.props.handleDataChange(this.state.value);
  }

  public render() {
    return (
      <input
        id={this.props.id}
        type={this.props.type}
        onBlur={this.onDataChangeSubmit}
        onChange={this.onDataChanged}
        disabled={this.props.readOnly}
        required={this.props.required}
        className={classNames('form-control',
          { 'validation-error': !this.props.isValid, 'disabled': this.props.readOnly },
        )}
        value={this.state.value}
      />
    );
  }
}
