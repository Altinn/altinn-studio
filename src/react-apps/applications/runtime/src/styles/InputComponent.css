.validation-error {
  border: 2px solid #E23B53 !Important;
}