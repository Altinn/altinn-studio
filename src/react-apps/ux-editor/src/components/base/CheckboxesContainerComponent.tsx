import * as React from 'react';

export interface ICheckboxContainerProps {
  id: string;
  component: IFormComponent;
  formData: any;
  handleDataChange: (value: any) => void;
  getTextResource: (resourceKey: string) => string;
  isValid: boolean;
  designMode: boolean;
}

export interface ICheckboxContainerState {
  selected: boolean;
}

export class CheckboxContainerComponent extends React.Component<ICheckboxContainerProps, ICheckboxContainerState> {

  public onDataChanged = (e: any) => {
    this.props.handleDataChange(!this.props.formData);
  }

  public render() {
    return (
      <div className={this.props.isValid ? 'form-group' : 'form-group validation-error'} id={this.props.id}>
        <div
          className={'pl-0 custom-control custom-control-stacked custom-checkbox a-custom-checkbox'}
          onClick={this.onDataChanged}
        >
          <input
            type='checkbox'
            checked={this.props.formData === true}
            name={this.props.component.id}
            className={this.props.isValid ? 'custom-control-input' : 'custom-control-input validation-error'}
          />
          <label className={'pl-3 custom-control-label a-fontBold'} htmlFor={this.props.component.id}>
            {this.props.designMode ? this.props.component.title : this.props.getTextResource(this.props.component.title)}
          </label>
        </div>
      </div>
    );
  }
}
