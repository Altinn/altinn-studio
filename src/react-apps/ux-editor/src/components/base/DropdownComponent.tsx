import * as React from 'react';

export interface IDropdownProps {
  id: string;
  component: IFormComponent;
  formData: any;
  handleDataChange: (value: any) => void;
  isValid?: boolean;
}

export interface IDropdownState {
  title: string;
  options: IOptions[];
  name: string;
}

export class DropdownComponent
  extends React.Component<IDropdownProps, IDropdownState> {

  public onDataChanged = (e: any) => {
    this.props.handleDataChange(e.target.value);
  }

  public render() {
    const { options } = this.props.component;
    return (
      <select
        id={this.props.id}
        value={this.props.formData}
        className={this.props.isValid ? 'custom-select a-custom-select' : 'custom-select a-custom-select validation-error'}
        onChange={this.onDataChanged}
      >
        {options.map((option, index) => (
          <option key={index} value={option.value}>
            {option.label}
          </option>
        ))}
      </select>
    );
  }
}
