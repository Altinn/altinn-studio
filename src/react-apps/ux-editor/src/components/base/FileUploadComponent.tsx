import * as React from 'react';

export interface IFileUploadProps {
  id: string;
  component: IFormComponent;
  formData: any;
  handleDataChange: (value: any) => void;
  isValid?: boolean;
}

export interface IFileUploadState {
  title: string;
  component: string;
  description: string;
}

export class FileUploadComponent
  extends React.Component<IFileUploadProps, IFileUploadState> {

  public onDataChanged = (e: any) => {
    this.props.handleDataChange(e.target.value);
  }

  public render() {
    const { description } = this.props.component;
    return (
      <div className='js-attachmentForm'>
        <label className='a-btn a-btn-action a-iconRight a-custom-fileupload mb-0 mt-1 a-js-uploadAttachment'>
          <input
            id={this.props.id}
            type='file'
            name='file'
            onChange={this.onDataChanged}
            className={this.props.isValid ? 'a-js-certificateContainer sr-only' : 'a-js-certificateContainer sr-only validation-error'}
          />
          Add a file
          {description}
          <i className='ai ai-upload' />
        </label>
      </div>
    );
  }
}
