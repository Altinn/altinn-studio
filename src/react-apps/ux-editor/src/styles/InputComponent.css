.validation-error {
  border: 2px solid red !Important;
}