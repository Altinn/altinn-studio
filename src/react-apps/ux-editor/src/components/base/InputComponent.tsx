import * as React from 'react';
import '../../styles/InputComponent.css';

export interface IInputProps {
  id: string;
  component: IFormComponent;
  formData: any;
  handleDataChange: (value: any) => void;
  isValid?: boolean;
}

export interface IInputState {
  value: string;
}

export class InputComponent
  extends React.Component<IInputProps, IInputState> {

  constructor(_props: IInputProps, _state: IInputState) {
    super(_props, _state);

    this.state = {
      value: _props.formData ? _props.formData : '',
    };
  }

  public onDataChanged = (e: any) => {
    this.setState({
      value: e.target.value,
    });
  }

  public onDataChangeSubmit = () => {
    this.props.handleDataChange(this.state.value);
  }

  public render() {
    return (
      <input
        id={this.props.id}
        type={this.props.component.type as any}
        onBlur={this.onDataChangeSubmit}
        onChange={this.onDataChanged}
        disabled={this.props.component.disabled}
        required={this.props.component.required}
        className={this.props.isValid ? 'form-control' : 'form-control validation-error'}
        value={this.state.value}
      />
    );
  }
}
