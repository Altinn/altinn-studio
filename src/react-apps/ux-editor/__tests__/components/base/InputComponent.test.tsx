import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { InputComponent } from '../../../src/components/base/InputComponent';

describe('>>> components/base/InputComponent.tsx --- Snapshot', () => {
  let mockId: string;
  let mockComponent: any;
  let mockFormData: any;
  let mockHandleDataChange: (value: any) => void;
  let mockIsValid: boolean;

  beforeEach(() => {
    mockId = "mock-id";
    mockComponent = {
      id: mockId,
      title: "test-checkboxescontainer",
      component: "Checkboxes",
    };
    mockHandleDataChange = (data: any) => null;
    mockIsValid = true;
  });

  it('>>> Capture snapshot of InputComponent', () => {
    const rendered = renderer.create(
      <InputComponent
        id={mockId}
        component={mockComponent}
        formData={mockFormData}
        handleDataChange={mockHandleDataChange}
        isValid={mockIsValid}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});