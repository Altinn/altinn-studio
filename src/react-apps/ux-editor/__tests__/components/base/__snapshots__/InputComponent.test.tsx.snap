// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> components/base/InputComponent.tsx --- Snapshot >>> Capture snapshot of InputComponent 1`] = `
<input
  className="form-control"
  id="mock-id"
  onBlur={[Function]}
  onChange={[Function]}
  value=""
/>
`;
